packages/untp-test-suite/src/interfaces/utils/common.ts

```diff
@@ -9,4 +18,50 @@ export async function readJsonFile<T>(filePath: string): Promise<T | null> {
   } catch (error) {
     return null;
   }
-}
\ No newline at end of file
+}
+
+export const getTemplateName = (testSuiteResult: IValidatedCredentials): string => {
+  if (testSuiteResult.errors === null) {
+    return TemplateName.pass;
+  }
+
+  if (Array.isArray(testSuiteResult.errors) && testSuiteResult.errors.length > 0) {
+    const getErrors = testSuiteResult.errors as ErrorObject[];
+    const errorWithAdditionalProperty = getErrors.find((error) => error?.params?.additionalProperty);
+    if (errorWithAdditionalProperty) {
+      // If any ErrorObject has 'additionalProperty', return 'warning'
+      return TemplateName.warn;
+    }
+
+    // If no ErrorObject has 'additionalProperty', return 'failed'
+    return TemplateName.failed;
+  }
+
+  return TemplateName.failed;
+};
+
+export const generateFinalMessage = (credentials: ICredentialTestResult[]): IFinalReport => {
+  let finalMessage = TestSuiteMessage.Pass;
+
+  if (credentials.some((credential) => credential.result === TemplateName.failed)) {
+    finalMessage = TestSuiteMessage.Failed;
+  }
+
+  if (credentials.some((credential) => credential.result === TemplateName.warn)) {
+    finalMessage = TestSuiteMessage.Warning;
+  }
+
+  let finalStatus: FinalStatus = 'PASS';
+  if (finalMessage === TestSuiteMessage.Failed) {
+    finalStatus = 'FAILED';
+  }
+
+  if (finalMessage === TestSuiteMessage.Warning) {
+    finalStatus = 'WARN';
+  }
```

Comment:
should merge the condition expressions for `finalStatus` and `finalMessage`
And the condition expression for failed should be at the last condition, because if we have 3 credentials and the result of the test is [pass, failed, warning], then final status based on this code will be `'WARN'`, it should be failed